lib/PTML/Emitter.cpp

```diff
@@ -0,0 +1,135 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/PTML/Constants.h"
+#include "revng/PTML/Emitter.h"
+
+using namespace ptml;
+
+static constexpr llvm::StringRef IndentString = "  ";
+
+template<bool EscapeQuotes = false>
+static bool requiresEscaping(char Character) {
+  switch (Character) {
+  case '<':
+  case '>':
+  case '&':
+    return true;
+  case '\"':
+    return EscapeQuotes;
+  default:
+    return false;
+  }
+}
+
+static llvm::StringRef getEscape(char Character) {
+  switch (Character) {
+  case '\"':
+    return "&quot;";
+  case '\'':
+    return "&apot;";
+  case '<':
+    return "&lt;";
+  case '>':
+    return "&gt;";
+  case '&':
+    return "&amp;";
+  default:
+    revng_abort("The specified character does not require escaping.");
+  }
+}
```

Comment:
Are you sure there are no other things we have to escape?
Why reinvent the wheel if we have `llvm::printHTMLEscaped` and `llvm::PrintEscapedString`?

Reply:
Sorry, I'm not following. Perhaps I'm missing some context. Can you clarify?